itest/rust/src/object_tests/dyn_gd_test.rs

```diff
@@ -379,6 +485,14 @@ trait Health {
     }
 }
 
+trait HealthWithAssociatedType {
+    type HealthType;
+
+    fn get_report(&self) -> Self::HealthType;
+
+    fn set_report(&mut self, new: Self::HealthType);
+}
```

Comment:
Maybe you can instead extend the `InstanceIdProvider`, e.g. like this?

```rs
trait IdProvider {
    type Id;
    fn get_id_dynamic(&self) -> Self::Id;
}
```

That would require less changes and keep the two traits easier to distinguish.

(Also, if you move it to its original declaration location, the diff is easier to inspect).